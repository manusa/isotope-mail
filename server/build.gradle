buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
