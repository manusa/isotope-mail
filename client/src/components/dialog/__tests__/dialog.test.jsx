import React from 'react';
import {shallow} from 'enzyme/build/index';
import Dialog from '../dialog';

describe('Dialog component test suite', () => {
  test('Snapshot render, should render dialog', () => {
    const {visible, title, actions, className, containerClassName, contentClassName} = {
      visible: true,
      title: 'Dr. POTUS',
      actions: [
        {label: 'Red label', action: () => {}},
        {label: 'Black disabled label', disabled: true, action: () => {}}
      ],
      className: 'first-class', containerClassName: 'container-class', contentClassName: 'content-class'};
    const dialog = shallow(<Dialog visible={visible} title={title} actions={actions}
      className={className} containerClassName={containerClassName} contentClassName={contentClassName}>
      <p>Dialog content</p>
    </Dialog>);
    expect(dialog).toMatchSnapshot();
  });
});
