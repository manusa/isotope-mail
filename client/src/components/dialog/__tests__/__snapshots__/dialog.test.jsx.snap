// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component test suite Snapshot render, should render dialog 1`] = `
<Portal
  containerInfo={
    <div
      id="modal-root"
    />
  }
>
  <div
    aria-modal="true"
    className="mdc-dialog first-class mdc-dialog--open"
    role="alertdialog"
  >
    <div
      className="mdc-dialog__container container-class"
    >
      <div
        className="mdc-dialog__surface"
      >
        <h2
          className="mdc-dialog__title"
        >
          Dr. POTUS
        </h2>
        <div
          className="mdc-dialog__content content-class"
        >
          <p>
            Dialog content
          </p>
        </div>
        <footer
          className="mdc-dialog__actions"
        >
          <Button
            className=" mdc-dialog__button"
            disabled={false}
            icon=""
            key="0"
            label="Red label"
            onClick={[Function]}
            type="button"
          />
          <Button
            className=" mdc-dialog__button"
            disabled={true}
            icon=""
            key="1"
            label="Black disabled label"
            onClick={[Function]}
            type="button"
          />
        </footer>
      </div>
    </div>
    <div
      className="mdc-dialog__scrim"
      onClick={[Function]}
    />
  </div>
</Portal>
`;
